docs/source/configuration_file.md

```diff
@@ -0,0 +1,139 @@
+- Other parameters that could be needed in the Loss function definition: see attributes of the Loss function of interest (e.g. `"gamma": 0.5` for `FocalLoss`).
+#### training_time
+- `num_epochs`: int.
+- `early_stopping_epsilon`: Threshold (percentage) for an improvement in the validation loss to be considered meaningful
+- `early_stopping_patience`: Number of epochs after which the training is stopped if the validation loss improvement not meaningful (i.e. less than `early_stopping_epsilon`)
+#### scheduler
+- `initial_lr`: Float. Initial learning rate.
+- `scheduler_lr`:
+1. `name`: Choice between: `"CosineAnnealingLR"`, `"CosineAnnealingWarmRestarts"` and `"CyclicLR"`. Please find documentation [here](https://pytorch.org/docs/stable/optim.html).
+2. Other parameters that are needed for the scheduler of interest (e.g. `"base_lr": 1e-5, "max_lr": 1e-2` for `"CosineAnnealingLR"`).
+#### balance_samples
+Bool. If `true`, then positive and negative GT samples are balanced in both training and validation datasets.
+#### mixup_alpha
+Float. Alpha parameter of the Beta distribution, see [original paper](https://arxiv.org/pdf/1710.09412.pdf).
+#### transfer_learning
+- `retrain_model`: Filename of the pretrained model (`path/to/pretrained-model`). If `null`, then no transfer learning is performed and the network is trained from scratch.
+- `retrain_fraction`: Float between 0. and 1. Controls the fraction of the pre-trained model that will be fine-tuned. For instance, if set to 0.5, then the second half of the model will be fine-tuned, while the first layers will be frozen.
+
+## Architecture
+Architectures for both segmentation and classification are available and described [here](models.rst).
+If the selected architecture is listed [here](https://github.com/neuropoly/ivado-medical-imaging/blob/master/ivadomed/loader/loader.py#L14), then a classification task is run, ie the ground-truth are labels extracted from `target`, instead of arrays for the segmentation task.
+### default_model (Mandatory)
+Define the default model (`Unet`) and mandatory parameters that are common to all available architectures (listed [here](models.rst)). If a tailored model is defined (see next section), then the default parameters are merged with the parameters that are specific to the tailored model.
+- `name`: `Unet` (default)
+- `dropout_rate`: float (e.g. 0.4).
+- `batch_norm_momentum`: float (e.g. 0.1).
+- `depth`: int, number of down-sampling operations.
+Note:
+- `in_channel` is automatically defined with `multichannel` and `contrast/training_validation`.
+- `out_channel` is automatically defined with `target_suffix`.
+### Tailored model (optional)
+Here are defined the tailored model and the parameters that are specific to it (ie not defined in the default model). See examples:
+- [FiLMedUnet](https://github.com/neuropoly/ivado-medical-imaging/blob/master/ivadomed/config/config.json#L64)
```

Comment:
even when not pointing to another file in the documentation (docs/), i would make a relative link. E.g.: if you create a branch where the json file changes, this absolute link to the master branch will be wrong

Reply:
Agree! Corrected :-)